### PR TITLE
MWPW-159791 Support Next-Gen Sidekick

### PR DESCRIPTION
**Pull request by meganthecoder**
* Adds support for [AEM Next-Gen Sidekick](https://chromewebstore.google.com/detail/aem-sidekick-nextgen/igkmdomcgoebiipaifhmpfjhbjccggml?pli=1) (AKA `<aem-sidekick>`)
* Backwards compatible with current AEM Sidekick (AKA `<helix-sidekick>`)

Resolves: [MWPW-159791](https://jira.corp.adobe.com/browse/MWPW-159791)

Note: The next-gen sidekick doesn't appear to have an options page, so "edit" and "prod" plugins can't be tested until after release.

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/methomas/send-to-caas?martech=off
- After: https://methomas-next-gen-sidekick--milo--adobecom.hlx.page/drafts/methomas/send-to-caas?martech=off

### CODE REVIEW

**Comment by meganthecoder** [comment]
AEM Sidekick - NextGen ([Chrome extension](https://chromewebstore.google.com/detail/aem-sidekick-nextgen/igkmdomcgoebiipaifhmpfjhbjccggml?pli=1))
Screenshot 2024-10-08 at 3 37 05 PM

**Review by Brandon32 — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Review by JasonHowellSlavin — APPROVED** [review]

File: libs/utils/sidekick.js [hunk 1]
```diff
@@ -32,13 +32,13 @@ export default function init({ createTag, loadBlock, loadScript, loadStyle }) {
...
+  const sk = document.querySelector('aem-sidekick, helix-sidekick');
 
   // Add plugin listeners here
   sk.addEventListener('custom:send-to-caas', sendToCaasListener);
   sk.addEventListener('custom:check-schema', checkSchemaListener);
   sk.addEventListener('custom:preflight', preflightListener);
 
   // Color code publish button
-  stylePublish(sk);
+  if (sk.nodeName === 'HELIX-SIDEKICK') stylePublish(sk);
```
[the remark this answers is not in the document]
Comment (meganthecoder) [reply libs/utils/sidekick.js:43]: @auniverseaway Thoughts on this? You said we shouldn't style the publish button, right? I could try to separate out the styling from the permissions functionality and only add the permissions to the next gen.

**Comment by meganthecoder** [comment]
I'm removing the Ready for Stage label until the question about `stylePublish` and permissions has been resolved.

File: libs/utils/sidekick.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (auniverseaway) [reply libs/utils/sidekick.js:43]: If we style the publish button today, it's probably fine to style it for the future. We run the same risks as today in that if the product changes, this feature may break... but again, we already have that risk today.

[the remark this answers is not in the document]
Comment (meganthecoder) [reply libs/utils/sidekick.js:43]: Sounds good! Created a follow-up ticket for this MWPW-160292

**Comment by meganthecoder** [comment]
> I'm removing the Ready for Stage label until the question about `stylePublish` and permissions has been resolved.

Added back label. Created follow up ticket [MWPW-160292](https://jira.corp.adobe.com/browse/MWPW-160292)

**Comment by auniverseaway** [comment]
Our agreement is to **_not_** colorize the publish button in favor of adding a confirmation dialog. Right, @meganthecoder?

**Review by Sartxi — APPROVED** [review]

**Comment by meganthecoder** [comment]
> Our agreement is to **_not_** colorize the publish button in favor of adding a confirmation dialog. Right, @meganthecoder?

  @auniverseaway That's correct

**Comment by mokimo** [comment]
@meganthecoder not blocking the PR but the preflight tool doesn't seem to work OOTB, on the first click there's a JS error, on the second one an error from within the preflight.
Screenshot 2024-10-23 at 15 58 51

Might make sense to open a separate story so this can be looked into as part of enabling the next gen sidekick

**Comment by meganthecoder** [comment]
> @meganthecoder not blocking the PR but the preflight tool doesn't seem to work OOTB, on the first click there's a JS error, on the second one an error from within the preflight.
> Might make sense to open a separate story so this can be looked into as part of enabling the next gen sidekick

@mokimo Thanks! I missed this since it looked like preflight was working correctly. Follow up ticket: https://jira.corp.adobe.com/browse/MWPW-160966

**Pull request merged**